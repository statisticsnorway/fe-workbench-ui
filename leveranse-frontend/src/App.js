import React from 'react'
import { Route } from 'react-router-dom'
import WelcomePage from './components/pages/WelcomePage'
import LoginPage from './components/pages/LoginPage'
import HomePage from './components/pages/HomePage'
import 'moment/locale/nb'

const App = () => (
  <div className='ui container'>
    <Route path='/' exact component={WelcomePage} />
    <Route path='/login' exact component={LoginPage} />
    <Route path='/home' exact component={HomePage} />
  </div>
)

export default App
