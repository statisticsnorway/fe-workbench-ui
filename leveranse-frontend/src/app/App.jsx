import React from 'react'
import { Router, Route } from 'react-router-dom'
import { history } from '../helpers'
import { PrivateRoute } from '../components'
import { HomePage } from '../pages/homePage'
import { LoginPage } from '../pages/loginPage'
import { RegisterPage } from '../pages/registerPage'
import WelcomePage from '../pages/WelcomePage'

class App extends React.Component {
  constructor (props) {
    super(props)
  }

  render () {
    const {alert} = this.props
    return (
      <div className="ui container">
        <div>
          <Router history={history}>
            <div>
              <PrivateRoute path="/home" exact component={HomePage}/>
              <Route path='/' exact component={WelcomePage}/>
              <Route path='/login' exact component={LoginPage}/>
            </div>
          </Router>
        </div>
      </div>
    )
  }
}

export default App