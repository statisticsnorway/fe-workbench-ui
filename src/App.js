import React, { Component } from 'react'

import { ContextProvider } from './context/ContextProvider'
import Home from './pages/home/Home'
import Login from './pages/login/Login'
import UserPreferences from './pages/userconfig/UserPreferences'

class App extends Component {
  state = {
    dataResource: [],
    loggedIn: false,
    role: '',
    user: '',
    preferencesSet: false,
    service: null
  }

  handleChange = (event, data) => {
    this.setState({ [data.name]: data.value })
  }

  handleUpdate = (event) => {
    event.preventDefault()

    this.setState({ loggedIn: true })
    // TODO get preferences from backend, set this.state.preferencesSet if present
  }

  handlePreferenceUpdate = (event) => {
    event.preventDefault()
    this.setState({preferencesSet: true})
  }

  handleLogout = () => {
    this.setState({
      loggedIn: false,
      preferencesSet: false
    })
  }

  render () {
    const { ldsURL } = this.props
    const { loggedIn, preferencesSet, ...user } = this.state

    if (!loggedIn) {
      return (
        <Login handleChange={this.handleChange} handleLogin={this.handleUpdate} {...this.state} />
      )
    } else if (!preferencesSet) {
      return (
        <ContextProvider>
          <UserPreferences handleChange={this.handleChange} handleUpdate={this.handlePreferenceUpdate} {...this.state} />
        </ContextProvider>
        )
    } else {
      return (
        <ContextProvider>
          <Home graphqlURL={ldsURL + 'graphql'} handleLogout={this.handleLogout} {...user} />
        </ContextProvider>
      )
    }
  }
}

export default App
