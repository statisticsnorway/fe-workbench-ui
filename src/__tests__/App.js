import React from 'react'
import { shallow } from 'enzyme'

import App from '../App'

describe('App', () => {
  it('Renders itself without error and only once', () => {
    const component = shallow(<App lds='http://localhost:9090/' />)

    expect(component.length).toEqual(1)
    expect(component).toMatchSnapshot()
  })
})
