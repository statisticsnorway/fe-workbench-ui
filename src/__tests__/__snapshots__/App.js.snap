// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders itself without error and only once 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    lds="http://localhost:9090/"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "basic": true,
      "children": <Container
        fluid={true}
        style={
          Object {
            "marginTop": "5em",
          }
        }
      >
        <Login
          handleChange={[Function]}
          handleLogin={[Function]}
          languageCode="en"
          loggedIn={false}
          password=""
          user=""
        />
      </Container>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Login
          handleChange={[Function]}
          handleLogin={[Function]}
          languageCode="en"
          loggedIn={false}
          password=""
          user=""
        />,
        "fluid": true,
        "style": Object {
          "marginTop": "5em",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleChange": [Function],
          "handleLogin": [Function],
          "languageCode": "en",
          "loggedIn": false,
          "password": "",
          "user": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "basic": true,
        "children": <Container
          fluid={true}
          style={
            Object {
              "marginTop": "5em",
            }
          }
        >
          <Login
            handleChange={[Function]}
            handleLogin={[Function]}
            languageCode="en"
            loggedIn={false}
            password=""
            user=""
          />
        </Container>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Login
            handleChange={[Function]}
            handleLogin={[Function]}
            languageCode="en"
            loggedIn={false}
            password=""
            user=""
          />,
          "fluid": true,
          "style": Object {
            "marginTop": "5em",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleChange": [Function],
            "handleLogin": [Function],
            "languageCode": "en",
            "loggedIn": false,
            "password": "",
            "user": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
